bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniPatientServiceImpl.java

```diff
@@ -83,6 +89,23 @@ public PatientConfigResponse getConfig() {
                 searchParameters.getFilterPatientsByLocation(), searchParameters.getFilterOnAllIdentifiers());
     }
 
+    @Override
+    public List<PatientResponse> searchSimilarPatients(PatientSearchParameters searchParameters, PatientResponseMapper patientResponseMapper) {
+        List<PatientResponse> patients = new ArrayList<PatientResponse>();
+        Set<Person> persons = personService.getSimilarPeople(searchParameters.getName(), null, searchParameters.getGender());
```

Comment:
We need to actually use Lucene here. OpenMRS's personService.getSimilarPeople does a handcrafted SQL query which doesn't perform well against large databases: https://github.com/openmrs/openmrs-core/blob/2.1.3/api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java#L82
(Also FYI not every Person is a Patient.)